### PR TITLE
kubelet: fix setup of emptyDir with sizeLimit (fsquota)

### LINKED ISSUE #117098: MountVolume.SetUp of emptyDir with sizeLimit fails on subsequent calls (fsquota)
### What happened?

When sizeLimit is specified subsequent invocations will fail with the message:

```
MountVolume.SetUp failed for volume "home" : requesting quota of different size: old -1 new 2Gi
```

### What did you expect to happen?

No error should be raised.

### How can we reproduce it (as minimally and precisely as possible)?

Create a pod with emptyDir and sizeLimit with FSQuotaMonitoring enabled, MountVolume.SetUp is called again only if the first iteration takes longer then the actual reconciler loop interval (100ms). 

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

version.Info{Major:"1", Minor:"27+", GitVersion:"v1.27.0-beta.0.87+330b5a2b8dbd68", GitCommit:"330b5a2b8dbd681811cb8235947557c99dd8e593", GitTreeState:"clean", BuildDate:"2023-04-10T09:05:56Z", GoVersion:"go1.20.2", Compiler:"gc", Platform:"linux/amd64"}

</details>

### Cloud provider

<details>

Self-managed cluster in Azure

</details>

### OS version

<details>

NAME="Red Hat Enterprise Linux"
VERSION="9.1 (Plow)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="9.1"

Linux test-master-0 6.3.0-rc3 #1 SMP PREEMPT_DYNAMIC Thu Apr  6 20:44:15 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

</details>

### Container runtime (CRI) and version (if applicable)

<details>

Containerd 1.6.15

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

CNI: CANAL 3.24.1

</details>

### PR DESCRIPTION
**Pull request by alex-matei**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
When size limit is specified subsequent invocations will fail because ibytes is changed to -1 and stored internally in quotaSizeMap during the first call.  Later invocation will see that the requested size doesn't match the actual stored value and it will fail.

#### Which issue(s) this PR fixes:
Fixes #117098

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs

```

### CODE REVIEW

**Comment by pacoxu** [comment]
The UT needs to be fixed and can we address the problem with a unit test or e2e test? Then we can avoid getting a regression for this issue.

**Comment by alex-matei** [comment]
@pacoxu Done, the test that was failing was exactly this case, calling AssignQuota with exactly the same size again, fixed it. I added a new test case for calling AssignQuota again with a different size, this one should fail all the time.

File: pkg/volume/util/fsquota/quota_linux_test.go [hunk 1]
```diff
@@ -461,6 +461,10 @@ var quotaTestCases = []quotaTestCase{
 	},
 	{
```
Comment (pacoxu) [inline pkg/volume/util/fsquota/quota_linux_test.go:462]: At a first glance of the failed ut, I did not realize that this test case is about setting a limitation smaller than the current quota.

Should we add a testName for the quotaTestCase struct to make it clear?

**Comment by pacoxu** [comment]
Logically, this LGTM. 
I need do some further test on it.

**Pull request merged**